src/overlays/actors/ovl_En_GeldB/z_en_geldb.c

```diff
@@ -33,93 +97,1466 @@ const ActorInit En_GeldB_InitVars = {
     (ActorFunc)EnGeldB_Update,
     (ActorFunc)EnGeldB_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A35310.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/EnGeldB_Init.s")
+static ColliderCylinderInit sBodyCylInit = {
+    { COLTYPE_UNK5, 0x00, 0x09, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x01, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 20, 50, 0, { 0, 0, 0 } },
+};
+
+static ColliderTrisItemInit sBlockTrisElementsInit[] = {
+    {
+        { 0x02, { 0x00000000, 0x00, 0x00 }, { 0xFFC1FFFF, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+        { { { -10.0f, 14.0f, 2.0f }, { -10.0f, -6.0f, 2.0f }, { 9.0f, 14.0f, 2.0f } } },
+    },
+    {
+        { 0x02, { 0x00000000, 0x00, 0x00 }, { 0xFFC1FFFF, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+        { { { -10.0f, -6.0f, 2.0f }, { 9.0f, -6.0f, 2.0f }, { 9.0f, 14.0f, 2.0f } } },
+    },
+};
+
+static ColliderTrisInit sBlockTrisInit = {
+    { COLTYPE_METAL_SHIELD, 0x00, 0x0D, 0x00, 0x00, COLSHAPE_TRIS },
+    2,
+    sBlockTrisElementsInit,
+};
+
+static ColliderQuadInit sSwordQuadinit = {
+    { COLTYPE_UNK10, 0x11, 0x00, 0x00, 0x00, COLSHAPE_QUAD },
+    { 0x00, { 0xFFCFFFFF, 0x00, 0x08 }, { 0x00000000, 0x00, 0x00 }, 0x81, 0x00, 0x00 },
+    { { { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f } } },
+};
+
+static DamageTable sDamageTable = {
+    0x10, 0x02, 0x01, 0x02, 0x10, 0x02, 0x02, 0x10, 0x01, 0x02, 0x04, 0x02, 0xF2, 0x02, 0x02, 0x02,
+    0x02, 0xE4, 0x60, 0xD3, 0x00, 0x00, 0x01, 0x04, 0x02, 0x02, 0x08, 0x04, 0x04, 0x00, 0x04, 0x00,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(unk_4C, 2000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 10, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -3000, ICHAIN_STOP),
+};
+
+static Vec3f sUnusedOffset = { 1100.0f, -700.0f, 0.0f };
+
+void EnGeldB_SetupAction(EnGeldB* this, EnGeldBActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnGeldB_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EffectBlureInit1 blureInit;
+    EnGeldB* this = THIS;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    thisx->colChkInfo.damageTable = &sDamageTable;
+    ActorShape_Init(&thisx->shape, 0.0f, ActorShadow_DrawFunc_Teardrop, 0.0f);
+    this->actor.colChkInfo.mass = 0xFE;
+    thisx->colChkInfo.health = 20;
+    thisx->colChkInfo.unk_10 = 50;
+    thisx->colChkInfo.unk_12 = 100;
+    thisx->naviEnemyId = 0x54;
+    this->keyFlag = thisx->params & 0xFF00;
+    thisx->params &= 0xFF;
+    this->blinkState = 0;
+    this->unkFloat = 10.0f;
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600A458, &D_0600B6D4, this->jointTbl, this->morphTbl, 24);
```

Comment:
id stick to the old names for the tables until were ready to change all at once

Reply:
I'd prefer to future-proof. Makes my job easier later when I have to handle the conflicts.

Reply:
well we dont know if we're gonna stick with them. we havent had any other opinions on the names. i would rather it not be introduced until its time